src/items.rs

```diff
@@ -3143,7 +3183,13 @@ fn format_header(
     let mut result = String::with_capacity(128);
     let shape = Shape::indented(offset, context.config);
 
-    result.push_str(format_visibility(context, vis).trim());
+    let visibility = format_visibility(context, vis);
+    if context.config.version() == Version::Two {
+        // format_visibility doesn't have a trailing space in Version::Two
+        result.push_str(&visibility);
+    } else {
+        result.push_str(visibility.trim());
```

Comment:
It turns out that there were other places in the codebase that already dealt with trimming the trailing space coming back from `format_visibility`.